### DISCUSSION TITLE
Hold relay state on ESP32-C3 during reboot?

### DISCUSSION THREAD

**Discussion by psitem** [arendst/Tasmota#17157] [Q&A] [answered]
I recently converted some [Switchbot mini plugs](https://templates.blakadder.com/switchbot_plugmini_W1901400.html) to Tasmota and noticed that the relay doesn't stay on while Tasmota restarts (warm boot). Having `PowerOnState 3` configured restores the state but having it flip off-on is undesirable.

Checking on other devices I have, a [CloudFree Light Switch](https://templates.blakadder.com/cloudfree_SW1.html) (also ESP32-C3) exhibits the same behavior, while an [MJ-S01](https://templates.blakadder.com/martin-jerry-MJ-S01.html), [Sonoff S31 Lite](https://templates.blakadder.com/sonoff_S31-lite.html), and [SWB1 power strip](https://templates.blakadder.com/swb1.html) do not (all ESP8266).

On the Switchbot I tried a number of different builds, the SwitchBOTA initially installed 11.x and I've tried the official 12.2 tasmota32c3.bin, the 12.2.05 dev, and a few other random builds. All behave the same.

Is this an ESP32-C3 specific issue? Can the initialization behavior be adjusted in the source code?

**Comment by Jason2866** [comment, marked as answer]
It depends on the choosen GPIO. If it is a GPIO with restrictions, it does flip at restart. Nothing Tasmota can change. Looks like GPIO 12 is used for driving the relay. This will end in a flip when a reset is done when no extern Pull up or Pull down (depends on design) is there.
The gpio is hardcoded to input enabled, pull-up resistor enabled (IE = 1, WPU = 1) until software takes control over.

**Reply by psitem** [reply]
How about for the Switchbot on GPIO6?

**Reply by Jason2866** [reply]
Can work, but if there is a extern resistor which pulls in the opposite "direction" there is a flicker too.
You can try if you do an OTA update, in state relay off. If you hear nothing during restart, the design is propably so choosen to hold the relay in off state when no signal is there from the esp32-c3.

**Reply by psitem** [reply]
No dice on that one too. Ok, thanks for the help!
